bpf/go_sql.c

```diff
@@ -0,0 +1,91 @@
+// Copyright The OpenTelemetry Authors
```

Comment:
It is based on their code, just heavily refactored.  I'm of the understanding that we need to retain this, as it could still be considered a derivative work.  What is the proper thing to do here?

Reply:
I guess I'll leave it up to your judgement then. If it's using some of their code then it should include it. After my first review of this code, I thought that writing something from scratch using our data structures would be much simpler. https://github.com/grafana/beyla/pull/298#discussion_r1330549512